### PR TITLE
#735 Support installer for global installation

### PR DESCRIPTION
**Pull request by CREITZ25**
#735 Support installer for global installation

changes in file functions to handle msi and exe files;
activate doUnzip for function doInstall

### CODE REVIEW

File: scripts/src/main/resources/scripts/functions [hunk 1]
```diff
@@ -729,14 +740,22 @@ function doReplaceExtractedFile() {
   fi
   if [ "${target}" = "${DEVON_IDE_HOME}/scripts" ] && doIsWindows
   then
-    (sleep 10;doBackup "${target}" "${3}";doRunCommand "mv '${1}' '${target}'") &
+    # Extension fo msi and exe files
+    if [ "${ext}" = ".msi" ] || [ "${ext}" != ".exe" ]
+    then
+      (sleep 10;doBackup "${target}" "${3}";doRunCommand "mv '${1}' '${target}'") &
+    fi
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/functions:747]: I do not see how this relates to msi and exe.
We can have a separate chat about this crazy `(sleep 10;...)` code.
It is only a workaround for windows file locks when devonfw-ide updates itself in case you call `devon ide update scripts`.
IMHO you should revert this change here (line 732/743)

**Review by hohwille — CHANGES_REQUESTED** [review]
@CREITZ25 thanks for your PR and taking up this story. You perfectly found the correct spots where to do the required changes. 👍 
I left some review comments for you to have a look. Thanks in advance.

File: scripts/src/main/resources/scripts/functions [hunk 2]
```diff
@@ -654,7 +654,9 @@ function doExtract() {
     doRunCommand "tar xf '${1}' -C '${target_dir}'"
   elif [ "${ext}" = ".zip" ] || [ "${ext}" = ".jar" ]
   then
-    doRunCommand "unzip -qn '${1}' -d '${target_dir}'"
+    # activate doUnzip
+    #doRunCommand "unzip -qn '${1}' -d '${target_dir}'"
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/functions:658]: done

File: scripts/src/main/resources/scripts/functions [hunk 3]
```diff
@@ -1161,7 +1180,9 @@ function doInstall() {
     then
       echo "${version}" > "${version_file}"
     fi
-    doEcho "Successfully installed ${software}"
+    # Extension fo msi and exe files
+    #doEcho "Successfully installed ${software}"
+    doEcho "Successfully installed ${software} or installation software for ${software}"
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/functions:1199]: It is not possible to determine whether it is installation software based on the file extension. This doesn't work on Linux. You would have to pass this information to doInstall as a parameter and take it into account in some functions. But that makes things confusing. An adapted message, maybe not as I phrased it, would make more sense from my point of view

[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/functions:1199]: What about this:
doEcho "Successfully installed (installation software for) ${software}"

File: scripts/src/main/resources/scripts/functions [hunk 4]
```diff
@@ -666,10 +668,17 @@ function doExtract() {
     then
       rm "${target_dir}/Applications"
     fi
+  # Extension fo msi and exe files
+  elif [ "${ext}" = ".msi" ] || [ "${ext}" != ".exe" ]
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/functions]: I corrected this error

File: scripts/src/main/resources/scripts/functions [hunk 5]
```diff
@@ -666,10 +668,17 @@ function doExtract() {
...
+  then
+    doEcho "move installation software"
+    doRunCommand "mv ${1} ${target_dir}"
   else
     doFail "Unknown archive format: ${ext}. Can not extract ${1}"
   fi
-  doEcho "Successfully extracted archive ${filename} to updates/extracted"
+  # Extension fo msi and exe files
+  #doEcho "Successfully extracted archive ${filename} to updates/extracted"
+  doEcho "Successfully extracted/moved archive ${filename} to updates/extracted"
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/functions]: done

File: scripts/src/main/resources/scripts/functions (hunk 1 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/functions]: I change this part, but we have to look again at it

File: scripts/src/main/resources/scripts/functions [hunk 6]
```diff
@@ -711,6 +714,13 @@ function doUnzip() {
   doRunCommand "unzip -qn '${1}' -d '${target_dir}'"
 }
 
+# $1: filename (e.g. ${filename})
+function doIsInstallerExtension()
+{
+  local myext="${1/*\./.}"
+  return $([ "${myext}" = ".msi" ] || [ "${myext}" = ".exe" ] || [ "${myext}" = ".pkg" ] || [ "${myext}" = ".bat" ])
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/functions:721]: The syntax `$(...)` is causing that the "command" between the parentheses gets executed.
The expression in the parentheses here will be evaluated to something like `0` (true/success) or e.g. `1` or `255` (false/failure).
Therefore you IMHO have to remove this extra syntax or this can not work as expected.

**Review by hohwille — CHANGES_REQUESTED** [review]
@CREITZ25 I found one last issue that IMHO needs to be fixed.
Everything else looks perfect now.

**Review by hohwille — APPROVED** [review]
@CREITZ25 to speed up, I simply fixed the minimal review issue I left so we can now merge.
